var/spack/repos/builtin/packages/py-pymol/package.py

```diff
@@ -13,19 +13,28 @@ class PyPymol(PythonPackage):
        movies. Open Source PyMOL is free to everyone!"""
 
     homepage = "https://pymol.org"
-    url      = "https://github.com/schrodinger/pymol-open-source/archive/v2.3.0.tar.gz"
+    url      = "https://github.com/schrodinger/pymol-open-source/archive/v2.4.0.tar.gz"
 
-    version('2.3.0', sha256='62aa21fafd1db805c876f89466e47513809f8198395e1f00a5f5cc40d6f40ed0')
+    version('2.4.0', sha256='5ede4ce2e8f53713c5ee64f5905b2d29bf01e4391da7e536ce8909d6b9116581')
 
     depends_on('python+tkinter', type=('build', 'run'))
     depends_on('freetype', type=('build', 'run'))
-    depends_on('glew')
-    depends_on('glm')
-    depends_on('freeglut', type=('build', 'run'))
+    depends_on('glew', type=('build'))
+    depends_on('glm', type=('build'))
     depends_on('py-numpy', type=('build', 'run'))
     depends_on('tcsh', type=('build', 'run'))
     depends_on('py-pyqt5', type=('build', 'run'))
     depends_on('py-pmw', type=('build', 'run'))
+    depends_on('libmmtf-cpp', type=('build', 'run', 'link'))
+    depends_on('msgpack-c', type=('build', 'run'))
+    depends_on('libpng@1.5.13', type=('build', 'run'))
```

Comment:
Is this the only version of libpng that works?

Reply:
It crashes with the default version of libpng when I was testing it. It kept giving a warning claiming that it was built with 1.6.37, but running 1.5.13, followed by an error about an ill formed image and the pymol app crashing. 

I tried setting libpng@1.6.37 as build, run, link, etc. but couldn't get that to work.

Reply:
Hmm, I'm guessing what happened is that you built it with Spack's 1.6.37, but you have 1.5.13 installed on the system, and when you ran it it picked up the wrong library. Was `LD_LIBRARY_PATH` set when you tried running it? Python libraries don't always like RPATH.

Reply:
That was my guess as well, but haven't been able to confirm that yet. I ran it by way of 

spack module tcl loads --dependencies  py-pymol

and then loading all the modules that way.

For some reason it stopped working when I went to try and test it again just now...

Will see if I can convince it to use libpng 1.6.37 at runtime